scripts/run_acceptance_tests.py

```diff
@@ -0,0 +1,197 @@
+
+from core.constants import constants  # isort:skip
+from scripts import build  # isort:skip
+from scripts import servers  # isort:skip
+
+GOOGLE_APP_ENGINE_PORT: Final = 8181
+ELASTICSEARCH_SERVER_PORT: Final = 9200
+PORTS_USED_BY_OPPIA_PROCESSES: Final = [
+    GOOGLE_APP_ENGINE_PORT,
+    ELASTICSEARCH_SERVER_PORT,
+]
+
+_PARSER: Final = argparse.ArgumentParser(
+    description="""
+Run this script from the oppia root folder:
+   python -m scripts.run_acceptance_tests
+
+The root folder MUST be named 'oppia'.
+""")
+
+_PARSER.add_argument(
+    '--skip-build',
+    help='If true, skips building files. The default value is false.',
+    action='store_true')
+_PARSER.add_argument(
+    '--prod_env',
+    help='Run the tests in prod mode. Static resources are served from '
+         'build directory and use cache slugs.',
+    action='store_true')
+_PARSER.add_argument(
+    '--suite', default='full',
+    help='Performs test for different suites'
+         'For performing a full test, no argument is required.')
+_PARSER.add_argument(
+    '--server_log_level',
+    help='Sets the log level for the appengine server. The default value is '
+         'set to error.',
+    default='error',
+    choices=['critical', 'error', 'warning', 'info'])
+_PARSER.add_argument(
+    '--source_maps',
+    help='Build webpack with source maps.',
+    action='store_true')
+
+
+def run_webpack_compilation(source_maps: bool = False) -> None:
+    """Runs webpack compilation.
+
+    Args:
+        source_maps: bool. Whether to compile with source maps.
+    """
+    max_tries = 5
+    webpack_bundles_dir_name = 'webpack_bundles'
+
+    for _ in range(max_tries):
+        try:
+            managed_webpack_compiler = (
+                servers.managed_webpack_compiler(use_source_maps=source_maps))
+            with managed_webpack_compiler as proc:
+                proc.wait()
+        except subprocess.CalledProcessError as error:
+            print(error.output)
+            sys.exit(error.returncode)
+            return
+        if os.path.isdir(webpack_bundles_dir_name):
+            break
+    else:
+        # We didn't break out of the loop, meaning all attempts have failed.
+        print('Failed to complete webpack compilation, exiting...')
+        sys.exit(1)
+
+
+def build_js_files(dev_mode: bool, source_maps: bool = False) -> None:
+    """Build the javascript files.
+
+    Args:
+        dev_mode: bool. Represents whether to run the related commands in dev
+            mode.
+        source_maps: bool. Represents whether to use source maps while
+            building webpack.
+    """
+    if not dev_mode:
+        print('Generating files for production mode...')
+
+        build_args = ['--prod_env']
+        if source_maps:
+            build_args.append('--source_maps')
+        build.main(args=build_args)
+
+    else:
+        build.main(args=[])
+        common.run_ng_compilation()
+        run_webpack_compilation(source_maps=source_maps)
```

Comment:
Arent these also defined in e2e tests?

Reply:
@vojtechjelinek I tried removing them and placing them in the common.py but then they are giving circular import problems.

Reply:
@ashish-patwal Can you explain the reason for the circular import problems? What's the exact chain of dependencies that causes it?

Reply:
@seanlip Sure, i'll add the dependency chain of this circular import. It'll take some time because rn i am writing the backend tests that's why

Reply:
```
Traceback (most recent call last):
  File "/home/lucifer/.pyenv/versions/3.8.15/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/lucifer/.pyenv/versions/3.8.15/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/lucifer/Opensource/oppia-main/oppia/scripts/run_e2e_tests.py", line 30, in <module>
    from scripts import common  # isort:skip pylint: disable=wrong-import-position, unused-import
  File "/home/lucifer/Opensource/oppia-main/oppia/scripts/common.py", line 37, in <module>
    from scripts import build
  File "/home/lucifer/Opensource/oppia-main/oppia/scripts/build.py", line 33, in <module>
    from scripts import install_third_party_libs
  File "/home/lucifer/Opensource/oppia-main/oppia/scripts/install_third_party_libs.py", line 31, in <module>
    from . import install_third_party  # isort:skip  pylint: disable=wrong-import-position, wrong-import-order
  File "/home/lucifer/Opensource/oppia-main/oppia/scripts/install_third_party.py", line 46, in <module>
    common.require_cwd_to_be_oppia(allow_deploy_dir=True)
AttributeError: partially initialized module 'scripts.common' has no attribute 'require_cwd_to_be_oppia' (most likely due to a circular import)
```

This is the traceback. It goes from `run_e2e_tests.py` -> `common.py` -> `build.py` -> `install_third_party_libs.py` -> `install_third_party.py` -> `common.py` again.

Reply:
Thanks, I'll leave follow-up comments on your latest PR commit.